src/entt/core/type_traits.hpp

```diff
@@ -335,13 +335,14 @@ struct type_list_transform;
 
 /**
  * @brief Applies a given _function_ to a type list and generate a new list.
+ * @tparam Container Template type containing types.
  * @tparam Type Types provided by the type list.
  * @tparam Op Unary operation as template class with a type member named `type`.
  */
-template<typename... Type, template<typename...> class Op>
-struct type_list_transform<type_list<Type...>, Op> {
+template<template<typename...> class Container, typename... Type, template<typename...> class Op>
```

Comment:
I see the point of this one. However, as an user, I would be surprised to see the following to working properly:
```cpp
using type = type_list_transform_t<std::tuple<T, U, V>, my_op>;
```
I'd be even more surprised by the sheer amount of errors and subtleties it can throw at me.

Reply:
Well, how about adding then ``static_assert(std::is_base_of_v<type_list<Type...>, Container<Type...>>, "The provided List must be entt::type_list or a derived class.")`` or making the ``type`` alias SFINAE
``using type = std::enable_if_t<std::is_base_of_v<type_list<Type...>, Container<Type...>>, Container<typename Op<Type>::type...>>``